src/arcade/potts/env/location/PottsLocation.java

```diff
@@ -321,29 +336,21 @@ public Location split(MersenneTwisterFast random, ArrayList<Integer> offsets,
      * necessarily be balanced in size.
      *
      * @param random  the seeded random number generator
-     * @param offsets  the percentage offset in each direction for split point
-     * @param direction  the direction of the split
+     * @param plane  the plane of the split
      * @param probability  the probability to decide which split to return
      * @return  a location with the split voxels
      */
-    public Location split(MersenneTwisterFast random, ArrayList<Integer> offsets,
-                          Direction direction, Double probability) {
+    public Location split(MersenneTwisterFast random,
+                          Plane plane, Double probability) {
         // Initialize lists of split voxels
         ArrayList<Voxel> voxelsA = new ArrayList<>();
         ArrayList<Voxel> voxelsB = new ArrayList<>();
         
-        // Calculate split point.
-        Voxel splitpoint = offsets == null ? getCenter() : getOffset(offsets);
-        
-        // Perform the split along the specified direction
-        if (direction == null) {
-            direction = getDirection(random);
-        }
-        
-        splitVoxels(direction, voxels, voxelsA, voxelsB, splitpoint, random);
+        splitVoxels(plane, voxels, voxelsA, voxelsB, random);
         connectVoxels(voxelsA, voxelsB, this, random);
-        
-        if (offsets == null) {
+
+        Voxel center = getCenter();
```

Comment:
We should revisit how this check if being done, since you essentially call `getCenter` twice.

Reply:
The only way I can think of addressing this is adding a boolean flag to the split function parameters, which doesn't seem super svelte to me

Reply:
That's fair -- okay to leave as is!